src/include/duckdb/main/extension_helper.hpp

```diff
@@ -120,8 +120,8 @@ class ExtensionHelper {
 	static ExtensionUpdateResult UpdateExtension(ClientContext &context, const string &extension_name);
 
 	//! Get the extension directory base on the current config
-	static string ExtensionDirectory(ClientContext &context);
-	static string ExtensionDirectory(DatabaseInstance &db, FileSystem &fs);
+	static string ExtensionDirectory(ClientContext &context, bool create = true);
```

Comment:
For clarity can we add this functionality as a separate method - `GetExtensionDirectoryPath` - which only returns the path (i.e. runs the first part of the current `ExtensionDirectory` function)? Then `ExtensionDirectory` can call that method and do the creation.

Reply:
Sure